### PR TITLE
ddl: fix error code and error name

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR refines all `Error`s defined in `ddl/ddl.go`:

- Assign specific codes for each error to prevent `1105(Unknown)` error code.
- Assign code `8060` for all errors caused by invalid DDL object state, for example, `ErrInvalidTableState` and `ErrInvalidDatabaseState` shares error code `8060`.
- Assign code `8050` for most errors that caused by the operations that TiDB not supported yet, for example, `errRunMultiSchemaChanges` and `errCantDropColWithIndex`.

Related Parser PR: https://github.com/pingcap/parser/pull/630

### What is changed and how it works?
NA

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported variable/fields change

Side effects

 - Breaking backward compatibility

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @jackysp

File: ddl/ddl.go [hunk 1]
```diff
@@ -74,166 +74,166 @@ var (
...
 
 	// ErrDupKeyName returns for duplicated key name
-	ErrDupKeyName = terror.ClassDDL.New(codeDupKeyName, "duplicate key name")
+	ErrDupKeyName = terror.ClassDDL.New(mysql.ErrDupKeyName, mysql.MySQLErrName[mysql.ErrDupKeyName])
 	// ErrInvalidDBState returns for invalid database state.
-	ErrInvalidDBState = terror.ClassDDL.New(codeInvalidDBState, "invalid database state")
+	ErrInvalidDBState = terror.ClassDDL.New(mysql.ErrInvalidDDLState, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrInvalidDDLState], "database"))
 	// ErrInvalidTableState returns for invalid Table state.
-	ErrInvalidTableState = terror.ClassDDL.New(codeInvalidTableState, "invalid table state")
+	ErrInvalidTableState = terror.ClassDDL.New(mysql.ErrInvalidDDLState, fmt.Sprintf(mysql.MySQLErrName[mysql.ErrInvalidDDLState], "table"))
```
Comment (zimulala) [inline ddl/ddl.go]: Do we need to change the way it is used?
Now we use it like `ErrInvalidTableState.GenWithStack("invalid table state %v", tbInfo.State)`. I think we can use it like `ErrInvalidTableState.GenWithStackByArgs(...)` in this PR.

Comment (bb7133) [reply ddl/ddl.go]: Done, thanks

**Comment by bb7133** [comment]
mysql-test need to be updated as well

**Pull request merged**